inference-engine/src/transformations/src/transformations/serialize.cpp

```diff
@@ -0,0 +1,374 @@
+            .set_value(create_atribute_list(adapter).c_str());
+    }
+    void on_adapter(
+        const std::string& name,
+        ngraph::ValueAccessor<std::vector<std::string>>& adapter) override {
+        m_data.append_attribute(name.c_str())
+            .set_value(create_atribute_list(adapter).c_str());
+    }
+};
+
+const std::map<ngraph::Node*, int> create_layer_ids(const ngraph::Function& f) {
+    std::map<ngraph::Node*, int> layer_ids;
+    int id = 0;
+    for (auto node : f.get_ordered_ops()) {
+        layer_ids[node.get()] = id++;
+    }
+    return layer_ids;
+}
+
+const std::vector<Edge> create_edge_mapping(
+    const std::map<ngraph::Node*, int>& layer_ids, const ngraph::Function& f) {
+    std::vector<Edge> edges;
+    for (auto node : f.get_ordered_ops()) {
+        if (dynamic_cast<ngraph::op::Parameter*>(node.get()) != nullptr) {
+            continue;
+        }
+
+        for (auto i : node->inputs()) {
+            auto source_output = i.get_source_output();
+            auto source_node = source_output.get_node();
+            auto current_node = i.get_node();
+
+            NGRAPH_CHECK(layer_ids.find(source_node) != layer_ids.end(),
+                         "Internal error.");
+            NGRAPH_CHECK(layer_ids.find(current_node) != layer_ids.end(),
+                         "Internal error.");
+
+            Edge e{};
+            e.from_layer = layer_ids.find(source_node)->second;
+            e.from_port =
+                source_node->get_input_size() + source_output.get_index();
+            e.to_layer = layer_ids.find(current_node)->second;
+            e.to_port = i.get_index();
+            edges.push_back(e);
+        }
+    }
+    std::sort(begin(edges), end(edges),
+              [](const Edge& a, const Edge& b) -> bool {
+                  return a.from_layer < b.from_layer;
+              });
+    return edges;
+}
+
+// TODO: refactor to Vistor API when Constant will be supporting it
+ConstantAtributes dump_constant_data(std::vector<uint8_t>& bin,
+                                     const ngraph::op::Constant& c) {
+    NGRAPH_CHECK(c.get_output_partial_shape(0.).is_static(),
+                 "Dynamic shape of ", c.get_name(), " (", c.get_type_name(),
+                 ") ", "is not supported.");
+
+    ConstantAtributes attr;
+    const uint8_t* p = reinterpret_cast<const uint8_t*>(c.get_data_ptr());
+    attr.size = ngraph::shape_size(c.get_shape()) * c.get_element_type().size();
+    attr.offset = bin.size();
+    bin.insert(end(bin), p, p + attr.size);
+    return attr;
+}
+
+std::string get_opset_name(const ngraph::Node* n) {
+    auto opsets = std::array<std::reference_wrapper<const ngraph::OpSet>, 5>{
+        ngraph::get_opset1(), ngraph::get_opset2(), ngraph::get_opset3(),
+        ngraph::get_opset4(), ngraph::get_opset5()};
+
+    // return the oldest opset name where node type is present
+    for (int idx = 0; idx < opsets.size(); idx++) {
+        int number = idx + 1;
```

Comment:
```suggestion
        if (opsets[idx].get().contains_op_type(n)) {
            return "opset" + std::to_string(idx + 1);
        }
```

Reply:
Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/719e203b225fd8386b4852adc2e208ab759b00bf)